apps/rush/src/actions/VersionAction.ts

```diff
@@ -77,82 +87,60 @@ export default class VersionAction extends BaseRushAction {
       }
     }
 
-    this._versionManager = new VersionManager(this.rushConfiguration);
+    this._versionManager = new VersionManager(this.rushConfiguration, this._getUserEmail());
     if (this._ensureVersionPolicy.value) {
       const tempBranch: string = 'version/ensure-' + new Date().getTime();
-      const git: Git = new Git(this._targetBranch.value);
+      this._versionManager.ensure(this._versionPolicy.value, true);
 
-      const updatedPackages: Map<string, IPackageJson> = this._versionManager.ensure(this._versionPolicy.value);
-      if (updatedPackages) {
+      const updatedPackages: Map<string, IPackageJson> = this._versionManager.updatedProjects;
+      if (updatedPackages.size > 0) {
         console.log(`${updatedPackages.size} packages are getting updated.`);
-        this._updateFiles(updatedPackages);
-        // Make changes in temp branch.
-        git.checkout(tempBranch, true);
-
-        // Stage, commit, and push the changes to remote temp branch.
-        git.addChanges();
-        git.commit();
-        git.push(tempBranch);
-
-        // Now merge to target branch.
-        git.checkout(this._targetBranch.value);
-        git.pull();
-        git.merge(tempBranch);
-        git.push(this._targetBranch.value);
-        git.deleteBranch(tempBranch);
+        this._gitProcess(tempBranch);
       }
+    } else if (this._bumpVersion.value) {
+      const tempBranch: string = 'version/bump-' + new Date().getTime();
+      this._versionManager.bump(this._versionPolicy.value,
+        BumpType[this._overwriteBump.value],
```

Comment:
It will be undefined. Adding a validate method to avoid.

---
In reply to: [134068665](https://github.com/Microsoft/web-build-tools/pull/277#discussion_r134068665) [](ancestors = 134068665)